### DISCUSSION TITLE
Put value to masked array

### DISCUSSION THREAD

**Discussion by raffyjef** [ARM-DOE/pyart#1223] [Q&A] [answered]
Hi! When using min_rhv=0.8. it masked values less than 0.8, then assigns what value to the masked array? Is it possible to assign a value of 0 for masked arrays? 

I'm trying to use np.where(data == np.nan) = 0. But it doesn't work because values are not equal to nan.

**Comment by mgrover1** [comment]
Which function are you working with?

**Comment by raffyjef** [comment]
I'm using different functions like pyart.correct_calculate_attenuation, pyart.retrieve.est_rain_rate_z and others

**Comment by mgrover1** [comment, marked as answer]
Py-ART uses a masked array for the values, not a regular numpy array with np.nan values. To fill the values, you would need to use something like this function

https://numpy.org/doc/stable/reference/generated/numpy.ma.MaskedArray.filled.html

For example:

```python
rho_hv = radar.fields["rho_hv"]["data"]

rho_hv.filled(fill_value=0)
```

**Reply by mgrover1** [reply]
@raffyjef let me know if this approach works for you!

**Reply by raffyjef** [reply]
I tried using this code:

rain = pyart.retrieve.est_rain_rate_z(radar, alpha=0.0075, beta=0.8403, refl_field='final_corrected_reflectivity_horizontal', rr_field=None)
rain['data'].np.ma.MaskedArray.filled(fill_value=0)
radar.add_field('rain_rate', rain)

however, it gives an AttributeError: 'MaskedArray' object has no attribute 'np'

**Reply by mgrover1** [reply]
If you check the type of rain["data"], it is a numpy masked array. You do not need to add `np.ma.MaskedArray`... try using the snippet I included:

This sample code
```python
rho_hv = radar.fields["rho_hv"]["data"]

rho_hv.filled(fill_value=0)
```

And applied to your workflow:
```python
rain['data'] = rain["data"].filled(fill_value=0)
```

**Reply by raffyjef** [reply]
Great! It worked. Thank you so much.
